app/controllers/order_items_controller.rb

```diff
@@ -6,26 +6,47 @@ def new
 
     def create
         @order_item = OrderItem.new(order_items_params)
```

Comment:
Let's simplify this! :)

You have if-else statement with no clear intention what is it used for.

let's have in our controller something like this:

```ruby


respond_to do |format|
  if save_new_or_existing_order_item(order_item_params)
    format.html { redirect_to root_path}
    format.json { render :new, status: :created}
  else
    format.html { redirect_to new_user_session_path, notice: "You must be signed-in first" }
    format.json { render json: @order_item.errors, status: :unprocessable_entity}
  end
end

```

Then in your helper class do somethinig like this:

```ruby
def save_new_or_existing_order_item order_item_params
  product_id = order_params[:product_id]
  if current_cart.order_items.include?(OrderItem.find_by_product_id(product_id))
            @existing_item = current_cart.order_items.find_by_product_id(product_id)
            @new_quantity = order_params[:quantity]
            @existing_item.quantity += @new_quantity
            @existing_item.save
  else
            OrderItem.new(order_items_params).save
  end
end
```

You'll probably need to modify this solution but you get the point

Reply:
Solved. I also added `update_existing_order_item` in the helper for quantity updating in the cart